akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala

```diff
@@ -140,3 +272,40 @@ object HttpCookie {
   private[http] val domainChars = ALPHANUM ++ ".-"
   private[http] val pathOrExtChars = VCHAR ++ ' ' -- ';'
 }
+
+/**
+ * The Cookie SameSite attribute as defined by <a href="https://tools.ietf.org/html/draft-ietf-httpbis-cookie-same-site-00">RFC6265bis</a>
+ * and <a href="https://tools.ietf.org/html/draft-west-cookie-incrementalism-00">Incrementally Better Cookies</a>.
+ */
+sealed trait SameSite extends Renderable {
+  def asJava: jm.headers.SameSite = this match {
+    case SameSite.Strict => jm.headers.SameSite.Strict
+    case SameSite.Lax    => jm.headers.SameSite.Lax
+    case SameSite.None   => jm.headers.SameSite.None
+  }
+
+  override private[http] def render[R <: Rendering](r: R): r.type = r ~~ (this match {
+    case SameSite.Strict => "Strict"
+    case SameSite.Lax    => "Lax"
+    case SameSite.None   => "None"
+  })
+}
+
+object SameSite {
+
+  def apply(s: String): Option[SameSite] = {
+    if ("Lax".equalsIgnoreCase(s)) Some(Lax)
+    else if ("Strict".equalsIgnoreCase(s)) Some(Strict)
+    else if ("None".equalsIgnoreCase(s)) Some(None)
+    else Option.empty
+  }
+
+  case object Strict extends SameSite
+  case object Lax extends SameSite
+
+  // SameSite.None is different from not adding the SameSite attribute in a cookie.
```

Comment:
Can be scaladoc and add link as mentioned above for the java version.

Reply:
I will mark this as resolved and, when we finish the conversation there, I will do the same for both APIs. ;-)